Code/GraphMol/CIPLabeler/CIPLabeler.cpp

```diff
@@ -163,21 +163,35 @@ void label(std::vector<std::unique_ptr<Configuration>> &configs) {
   }
 }
 
+thread_local unsigned int remainingCallCount=0;
+
 }  // namespace
 
 void assignCIPLabels(ROMol &mol, const boost::dynamic_bitset<> &atoms,
-                     const boost::dynamic_bitset<> &bonds) {
+                     const boost::dynamic_bitset<> &bonds, unsigned int maxRecursiveIterations) {
+
+  if (maxRecursiveIterations != 0)
+    remainingCallCount  = maxRecursiveIterations;
+  else
+    remainingCallCount = UINT_MAX;  // really big - will never be hit
+  
+
   CIPMol cipmol{mol};
   auto configs = findConfigs(cipmol, atoms, bonds);
   label(configs);
 }
 
-void assignCIPLabels(ROMol &mol) {
+void assignCIPLabels(ROMol &mol, unsigned int maxRecursiveIterations) {
   boost::dynamic_bitset<> atoms(mol.getNumAtoms());
   boost::dynamic_bitset<> bonds(mol.getNumBonds());
   atoms.set();
   bonds.set();
-  assignCIPLabels(mol, atoms, bonds);
+  assignCIPLabels(mol, atoms, bonds, maxRecursiveIterations);
+}
+
+bool decrementRemainingCallCountAndCheck()
+{
+  return (--remainingCallCount) >0 ;
```

Comment:
Code-formating consistency:
```suggestion
bool decrementRemainingCallCountAndCheck() {
  return (--remainingCallCount) >0 ;
```
if you run `clang-format` over the code pre-commit (or from your editor), it will fix stuff like this for you.